src/ir/proto/raksha_ir.proto

```diff
@@ -0,0 +1,101 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+
+// This proto file contains the proto representation of the Raksha IR. This
+// allows frontends to construct the Raksha IR in any language and send the IR
+// over the wire or across an FFI layer for analysis.
+
+syntax = "proto3";
+
+package raksha.ir.proto;
+
+message Module {
+  map<uint64, Block> blocks = 1;
+}
+
+message Block {
+  // A mapping from the ID for an operation in the block to that operation.
+  // This allows values produced by this `Operation` to "point" to the
+  // `Operation` that produced them within the block.
+  map<uint64, Operation> operations = 1;
+  NamedValueMap results = 2;
+}
+
+message AttributePayload {
+  oneof AttributePayloadKind {
+    int64 int64_payload = 1;
+    string string_payload = 2;
+  }
+}
+
+message NamedAttributeMap {
+  map<string, AttributePayload> attributes = 1;
+}
+
+message NamedValueMap {
+  map<string, Value> values = 1;
+}
+
+message Operation {
+  string operator = 1;
```

Comment:
Create an `Operator` message and make this a field of `Operator` type. I expect we will add some additional info (like number of inputs, is_variadic, etc.)

Reply:
I was thinking of going a different way with it: if we were to add more fields to `Operator`, it is highly likely that we would want those additional fields to be the same across all uses of the `Operator` (ie, every `+` operator has 2 inputs, is not variadic, etc). As such, I was thinking that we could, in the TU-level structure, have a list of `FullOperatorInfo` (or maybe just `Operator`) providing this additional info in a map keyed on the operator name.

I suppose that if we want to do that, the best thing to do would be to rename `operator` here to `operator_name` to make it clear that this isn't the whole `Operator` concept, but just the name piece of it.

Reply:
`operator` -> `operator_name` with a brief comment sounds good.

Reply:
Done.